core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -97,8 +97,8 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
             data.map(notificationHandler.processNotification(_)).getOrElse {
```

Comment:
zkUtils.readDataMaybeNull method can return   (Some(string), Stat)  or  (None, Stat). So in this case,  if we are receiving Some(null), then we should pass null to notificationHandler.  Pl correct me If I am wrong.

Updated PR with a simple test case.